[MVP-1519] Intercom chat window

marukohao opened the pull request:
Use mock data for now(will remove mock data in this pr when embed firing API):
Incoming messages:
Should be a gray bubble
Show username
Show time

Outgoing messages (messages from you):
Blue bubble
Should not show username
Show time

Send message:
The message should not have a character limit
It should be scrollable if it is a long message
The Send button should be gray when the user has nothing in the message box.
The Send button should be blue when the user has sent something in the message box.

will working on the alert setting page next, so the alert setting button is disable now.

Test:
update:
removed the outline for the send message input box
update font size and font color of input box placeholder
![Untitled](https://user-images.githubusercontent.com/26240001/201209158-12c05351-12d2-458b-9647-e66bfb79090b.gif)


![Untitled](https://user-images.githubusercontent.com/26240001/200683622-1331b288-23dd-4d1d-a238-87f2094568b0.gif)

Story: https://notifi.atlassian.net/browse/MVP-1519
https://notifi.atlassian.net/browse/MVP-1558
https://notifi.atlassian.net/browse/MVP-1564
https://notifi.atlassian.net/browse/MVP-1560
https://notifi.atlassian.net/browse/MVP-1563
https://notifi.atlassian.net/browse/MVP-1561

jennifer-notifi commented on packages/notifi-react-card/lib/components/subscription/defaults.css:
```diff
@@ -53,7 +54,16 @@
+  width: 364px;
+}
+
+.NotifiIntercomCard__chatWindowContainer {
```
We should add some padding between the message box and messages themselves. Could you add some padding? Not 100% sure if it would be here though haha

marukohao commented on packages/notifi-react-card/lib/components/intercom/NotifiIntercomCardContainer.tsx:
```diff
@@ -73,29 +79,46 @@ export const NotifiIntercomCardContainer: React.FC<
-        disabled={disabled}
-        classNames={classNames?.NotifiStartChatButton}
-      />
+      {startChat ? (
```
create NotifiConsumerFTU, and showing NotifiConsumerFTU when stats is 'fetched'. and moved all the logic for showing startChatView in the NotifiConsumerFTU.

jennifer-notifi replied:
I think how you have it now works as well. 

I was thinking something more along the lines of where each page or 'view' is tied to its own state. 

```
useEffect(() => {
  if (userHasConversationActive) 
     setEditState('conversationActive');
})

case 'fetched': 
   return StartChatView;
case 'conversationActive'
  return ChatWindowView;
```

You will find this helpful when you create the Alert Settings page so that you don't have a nested conditional on what you render.

Maybe look into: `packages/notifi-react-card/lib/components/subscription/SubscriptionCardV1.tsx`

marukohao replied:
updated

jennifer-notifi commented:
Hey Hao, can you please update the screenshot/video when you are done making the updates?

marukohao commented:
> Hey Hao, can you please update the screenshot/video when you are done making the updates?

updated with the screenshot to show the padding change on the message, added Virtuoso but haven
't done with the set up yet, will do and test more when connecting with backend API.

jennifer-notifi commented:
> > Hey Hao, can you please update the screenshot/video when you are done making the updates?
> 
> updated with the screenshot to show the padding change on the message, added Virtuoso but haven 't done with the set up yet, will do and test more when connecting with backend API.

The virtual container looks good (for now), you can always copy and paste the same mock data to make a long array to test.

Also if these are easy updates:
- Update the font of the placeholder
- If there is a way to increase the padding more between the message and message box. You will see that when you type something into the message box, the message overlaps a bit around the blue border of the message box.
